modules/vpc/output.tf

```
@@ -0,0 +1,11 @@
+output "infrastructure_subnets" {
+  value = {
+    for k, s in aws_subnet.aap_infrastructure_subnets : k => s.id
+  }
+
+  depends_on = [aws_subnet.aap_infrastructure_subnets]
+}
+
+output "infrastructure_igw" {
+  value = aws_internet_gateway.aap_infrastructure_igw.id
+}
```

```suggestion
}

output "deployment_id" {
  value = var.deployment_id == "" ? random_string.deployment_id[0].id : var.deployment_id
}

```
tflint wanted this output here instead of the root output.tf. Guessing because it's actually used / set in this module.

---

Getting error if I moved it inside modules/vpc/output.tf
```
╷
│ Error: Reference to undeclared resource
│ 
│   on modules/vpc/output.tf line 14, in output "deployment_id":
│   14:   value = var.deployment_id == "" ? random_string.deployment_id[0].id : var.deployment_id
│ 
│ A managed resource "random_string" "deployment_id" has not been declared in module.vpc.
```

---

Sorry, I updated the suggestion to fix that. There's another way though. Keep the deployment_id in the root outputs.tf and add a description makes this happy. That's after the file name changes to include the missing `s`.